server/partition.go

```diff
@@ -232,6 +236,34 @@ func (p *partition) Delete() error {
 	return p.stopLeadingOrFollowing()
 }
 
+// IncreaseSubscriberCount increases the number of subscribers. Partitions with
+// a subscriber count greater than zero will not be auto-paused if the partition
+// is idle, and the corresponding configuration option is set.
+func (p *partition) IncreaseSubscriberCount() {
+	p.mu.RLock()
+	defer p.mu.RUnlock()
+
+	if p.autoPauseDisableIfSubscribers {
```

Comment:
I think it's confusing to only set `subscriberCount` if `autoPauseDisableIfSubscribers` is  true. I would instead only read the subscriber count in the auto pause loop if `autoPauseDisableIfSubscribers` is true

Reply:
I know, I was trying to avoid having to either use a mutex in `autoPauseLoop` or use an int8 (as atomic bool doesn't exist). I'll push a change.

Reply:
Fixed in 1683b9f.

Reply:
Improved in 3686551.